enos/enos.py

```diff
@@ -182,6 +182,11 @@ def up(provider=None, env=None, **kwargs):
 def install_os(env=None, **kwargs):
     logging.debug('phase[os]: args=%s' % kwargs)
 
+    if 'provider_net' not in env:
```

Comment:
Even better (IMO), make use of a decorator to validate the env prior to the task execution.